### PR TITLE
feature copy results to clipboard

### PR DESCRIPTION
**Pull request by marmoure**
while working with eXide and running eval on some scripts i found that copying the results was not easy at all.
first i can't select it all using ctrl-A as the focus was always on the editor pane and trying to do it with the mouse proven to be a hassle.
i decided to add a copy to clipboard button which copies all the results showing in the results pane.
image

### CODE REVIEW

File: index.html.tmpl [hunk 1]
```diff
@@ -419,6 +419,9 @@
                                         <input type="checkbox" id="auto-expand-matches" name="auto-expand-matches" checked="checked"/>
                                         Highlight Index Matches
                                     </label>
+                                    <label id="copy-all-clipboard" title="copy the resutls into the clipboard">
+                                    <i class="fa fa-clipboard" aria-hidden="true"></i>copy to clipboard
+                                    </label>
```
Comment (joewiz) [inline index.html.tmpl]:
```suggestion
                                    <label id="copy-all-clipboard" title="Copy the displayed results into the clipboard">
                                    <i class="fa fa-clipboard" aria-hidden="true"></i> Copy to clipboard
                                    </label>
```

File: src/eXide.js [hunk 2]
```diff
@@ -1551,6 +1551,16 @@ eXide.app = (function(util) {
                 util.requestFullScreen(document.getElementById("fullscreen"));
             });
             $(".results-container .layout-switcher").click(app.switchResultsPanel);
+			$('.results-container #copy-all-clipboard').click(() => {
+                let res = '';
+                document.querySelectorAll("#results-body > div > div > div > div.item").forEach(item => res += item.innerText)
+                navigator.clipboard.writeText(res).then(function() {
+                    console.log('Async: Copying to clipboard was successful!');
+                    eXide.util.message("Copyed to clipboard");
```
Comment (joewiz) [inline src/eXide.js]:
```suggestion
                    eXide.util.message("Copied results to clipboard");
```

**Review by joewiz — APPROVED** [review]
It works great! Just a couple of suggestions in the wording. Thanks for adding tests too! I notice these appear to be failing?

**Comment by marmoure** [comment]
@joewiz yes seems like CI test for copying doesn't work in this environment.
the clipboard requires special permissions and i tried to work around that
but seems like it failed. 
can you please run the test locally

**Comment by joewiz** [comment]
@marmoure I also get the same "Browser context management is not supported" error shown in the CI log when running `npm run cypress`.

**Comment by marmoure** [comment]
after updating the test i m still seeing a failure for checking the notification which is there???!!
Navbar -- should display notification (failed)

**Comment by marmoure** [comment]
@joewiz this unfortunately needs a lot of work to fix the broken tests
it Fixes #499 
how should we progress?

**Comment by joewiz** [comment]
@marmoure Ah, I'm not sure... Is there a way to keep the tests while marking them as "pending" or "don't run"? Perhaps the situation with browser context management will change in the future, and we could just re-enable your tests.

Re: #499, this PR is indeed quite similar, but my original idea was to allow copying on a per-item basis, but yours copies the current 10 results as a whole. I don't suppose it would be hard to add a per-item copy function? I'm happy to merge this without that, but I thought I'd ask in case it helps.

**Comment by marmoure** [comment]
@joewiz can you please elaborate and how that function work When you have multiple results.

**Review by joewiz — APPROVED** [review]
As discussed on today's call, we will merge this as is.

@marmoure Thank you!

**Pull request merged**